### PR TITLE
framework: improve guest cluster deletion

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/3892

### CODE REVIEW

File: pkg/framework/host.go [hunk 1]
```diff
@@ -177,17 +178,76 @@ func (h *Host) CreateNamespace(ns string) error {
 	return nil
 }
 
-func (h *Host) DeleteGuestCluster(name, cr, logEntry string) error {
-	if err := runCmd(fmt.Sprintf("kubectl delete %s %s", cr, h.clusterID)); err != nil {
-		return microerror.Mask(err)
+func (h *Host) DeleteGuestCluster(ctx context.Context, provider string) error {
```
Comment (kopiczko) [inline pkg/framework/host.go]: I'm thinking about making `provider` a mandatory `Host` field. Is it a problem for any e2e test we run?

Comment (MarcelMue) [reply pkg/framework/host.go]: Kind of, we have e2e tests that don't need a guest cluster (e.g. operatorkit or chart-operator)

**Review by fgimenez — APPROVED** [review]
LGTM

**Review by tuommaki — APPROVED** [review]
:+1:

**Review by MarcelMue — APPROVED** [review]
Fine with me

**Review by TheoBrigitte — APPROVED** [review]
LGTM

**Comment by kopiczko** [comment]
Tested with: https://github.com/giantswarm/aws-operator/pull/1117
Workflow: https://circleci.com/workflow-run/5b1906e1-d6cb-443f-9fc4-50483be3c2f6

```
D 18-08-17T08:26:32.273628+0 failure count is 5.000000 | aws-operator/integration/test/draining/draining_test.go:223
D 18-08-17T08:26:32.273659+0 success count is 674.000000 | aws-operator/integration/test/draining/draining_test.go:224
D 18-08-17T08:26:32.273681+0 ration is 0.741840 of failures | aws-operator/integration/test/draining/draining_test.go:225
--- PASS: Test_Draining (537.61s)
PASS
D 18-08-17T08:26:32.273756+0 deleting CR for guest cluster `ci-wip-fb486-6b509` | e2e-harness/pkg/framework/host.go:183
D 18-08-17T08:26:32.280867+0 deleted CR for guest cluster `ci-wip-fb486-6b509` | e2e-harness/pkg/framework/host.go:211
D 18-08-17T08:26:32.280928+0 ensuring CR for guest cluster `ci-wip-fb486-6b509` does not exist | e2e-harness/pkg/framework/host.go:215
W 18-08-17T08:26:32.284928+0 retrying backoff in '456.938922ms' due to error | backoff/notifier.go:13
        /home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/framework/host.go:236: guest cluster `ci-wip-fb486-6b509` CR still exists
        cluster deletion error
W 18-08-17T08:26:32.744706+0 retrying backoff in '608.059382ms' due to error | backoff/notifier.go:13
        /home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/framework/host.go:236: guest cluster `ci-wip-fb486-6b509` CR still exists
        cluster deletion error
W 18-08-17T08:26:33.355604+0 retrying backoff in '658.232989ms' due to error | backoff/notifier.go:13
        /home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/framework/host.go:236: guest cluster `ci-wip-fb486-6b509` CR still exists
        cluster deletion error
W 18-08-17T08:26:34.017018+0 retrying backoff in '2.470509327s' due to error | backoff/notifier.go:13
        /home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/framework/host.go:236: guest cluster `ci-wip-fb486-6b509` CR still exists
        cluster deletion error
W 18-08-17T08:26:36.49058+0 retrying backoff in '3.066747407s' due to error | backoff/notifier.go:13
        /home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/framework/host.go:236: guest cluster `ci-wip-fb486-6b509` CR still exists
        cluster deletion error
W 18-08-17T08:26:39.55991+0 retrying backoff in '3.555659102s' due to error | backoff/notifier.go:13
        /home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/framework/host.go:236: guest cluster `ci-wip-fb486-6b509` CR still exists
        cluster deletion error
W 18-08-17T08:26:43.118393+0 retrying backoff in '7.365273664s' due to error | backoff/notifier.go:13
        /home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/framework/host.go:236: guest cluster `ci-wip-fb486-6b509` CR still exists
        cluster deletion error
W 18-08-17T08:26:50.486738+0 retrying backoff in '7.170594758s' due to error | backoff/notifier.go:13
        /home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/framework/host.go:236: guest cluster `ci-wip-fb486-6b509` CR still exists
        cluster deletion error
W 18-08-17T08:26:57.659982+0 retrying backoff in '14.005838539s' due to error | backoff/notifier.go:13
        /home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/framework/host.go:236: guest cluster `ci-wip-fb486-6b509` CR still exists
        cluster deletion error
D 18-08-17T08:27:11.66783+0 ensured CR for guest cluster `ci-wip-fb486-6b509` does not exist | e2e-harness/pkg/framework/host.go:247
I 18-08-17T08:27:11.752466+0 finished tests | e2e-harness/pkg/project/project.go:171
```

**Pull request merged**